vocabs/DID-core.jsonld

```diff
@@ -0,0 +1,144 @@
+{
+    "@context" : {
+        "isDefinedBy" : {
+          "@id" : "http://www.w3.org/2000/01/rdf-schema#isDefinedBy",
+          "@type" : "@id"
+        },
+        "comment" : {
+          "@id" : "http://www.w3.org/2000/01/rdf-schema#comment"
+        },
+        "sameAs" : {
+          "@id" : "http://www.w3.org/2002/07/owl#sameAs",
+          "@type" : "@id"
+        },
+        "label" : {
+          "@id" : "http://www.w3.org/2000/01/rdf-schema#label"
+        },
+        "domain" : {
+          "@id" : "http://www.w3.org/2000/01/rdf-schema#domain",
+          "@type" : "@id"
+        },
+        "range" : {
+          "@id" : "http://www.w3.org/2000/01/rdf-schema#range",
+          "@type" : "@id"
+        },
+        "sec" : "https://w3id.org/security#",
+        "as" : "https://www.w3.org/ns/activitystreams",
+        "rdf" : "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+        "owl" : "http://www.w3.org/2002/07/owl#",
+        "xsd" : "http://www.w3.org/2001/XMLSchema#",
+        "rdfs" : "http://www.w3.org/2000/01/rdf-schema#",
+        "did" : "https://w3id.org/did#"
+    },
+    
+    "@graph" : [ {
+      "@id" : "https://w3id.org/did#",
+      "@type" : "owl:Ontology"
+    }, {
+      "@id" : "did:DIDDocument",
```

Comment:
Sorry I'm late to review this. I'm a bit concerned about adding these classes. Especially the blurring of the DID Subject and DID Document here. Maybe it is completely harmless, but the RDF users in my experience would expect to see such terms documented in the spec or namespace document, and they won't be (and adding them is going to confuse folks who aren't interested in the RDF side of this, I'm sure).

I'm also not sure about the implications on inferencing of having these classes. We're not going to leave the door open for incorrect assertions to be made here, are we? You have probably thought all of this through though.

Reply:
By which I mean, by using `did:DIDDocument` as the domain of various properties, we are implying that the property values apply to the DID document, and not the DID subject, are we not? But I thought there was general agreement that there is 'no such thing' as a DID document, rather it is a serialisation of attributes which apply to the DID subject (which in some cases is a document, and in some cases is a person or a cat, etc). I don't want to re-open that debate though.

Reply:
Removing the classes from the vocabulary means

- We cannot properly define property domains and ranges, meaning that the vocabulary file would 'just' be a series of `rdf:property` statements. We can do that, but it becomes a bit meaningless. We can document the classes in the vocabulary itself, and making it clear why it is not in the formal spec... 

    Fair to say that I do not have the good background to judge whether, from a security/privacy point of view, there is any harm in these; I do not see any. @dlongley or @msporny should say that. But, as a general notion, if we do use JSON-LD, ie, we use RDF(S), then the open world assumption stands for DID Documents, too...

- I am not sure the SHACL would work correctly. The "shapes" are attached to classes (SHACL makes use of the vocabulary). At least the 'starting' one, namely DIDDocument class, is needed (ShEx does not have this problem because it expects an explicit "mapping")

Reply:
> By which I mean, by using did:DIDDocument as the domain of various properties, we are implying that the property values apply to the DID document, and not the DID subject, are we not?

Aha! Maybe there is a naming thing here, which is my mistake.

As you may see there is no such concept in the vocabulary as DID subject because that is 'just' the URL of the, well, subject. Which makes me realize that what I called the `DIDDocument` class should be named `DIDSubject`!

Ie, it should say:

```
did:verificationMethod a rdf:Property ;
    rdfs:range did:VerificationMethod ;
    rdfs:domain did:DIDSubject ;
    ....
```

which means that if I have data of the form

```
<did:ex:123456> did:verificationMethod <did:ex:5t5t5t5t#something>
```

Then the inference would say that `<did:ex:123456>` is of class `DIDSubject`. And this is what we want.

Bottom line: I made a mistake in naming.